code/__HELPERS/time.dm

```diff
@@ -97,4 +97,19 @@
 
 //returns timestamp in a sql and ISO 8601 friendly format
 /proc/SQLtime()
-	return time2text(world.realtime, "YYYY-MM-DD hh:mm:ss")
\ No newline at end of file
+	return time2text(world.realtime, "YYYY-MM-DD hh:mm:ss")
+
+//Returns time as a "slot", a predefined type of time, see dates.dm for defines
+/proc/getTimeslot()
+	switch(text2num(time2text(world.timeofday, "hh")))
+		if(3 to 11)
```

Comment:
Could do `#define SLEEPTIME_TIMESLOT 3 to 11` and then enter `if(SLEEPTIME_TIMESLOT)` if you ever want to re-use those timeslots elsewhere.

Reply:
is there a way to make this work with the LATE entry?

Reply:
Yes.
`#define LATE 20 to 23, 0` would work.